app/processing/process-payment-requests.js

```diff
@@ -23,7 +25,13 @@ const processPaymentRequest = async (scheduledPaymentRequest) => {
   for (const paymentRequest of paymentRequests) {
     await mapAccountCodes(paymentRequest)
   }
-  await completePaymentRequests(scheduledPaymentRequest.scheduleId, paymentRequests)
+
+  // If has AR but no debt enrichment data, then route to request editor and apply hold
+  if (requiresDebtData(paymentRequests)) {
```

Comment:
Only way to know if this needs to go to the request editor for recovery data, is if we do the post payment adjustment and work that out.

Therefore the approach I've taken is to do the calculation, but if a recovery is detected then instead of completing it, instead route the original request to `ffc-pay-request-editor` and apply a hold to processing agreements for this scheme and customer.

As the schedule is not completed, once the hold is removed, the calculation will automatically repeat and this condition will identify that the debt data now exists.

My original approach to solving this problem was to still complete the request but add a column to the completed table, `awaitingEnrichment`.  When the data was provided from `ffc-pay-request-editor` the data was added and the new column updated.  Only requests not awaiting enrichment would be released.

That worked fine, but the risk is that the calculation and ledger assignment is dependent on knowing the settlement position at a point in time.  If the settlement position changes whilst the request is in the editor then we'd be sending invalid data to DAX.

The potential issue with the new approach is that all marketing years will be held for SFI.  This may well be the intended behaviour as SFI is a multi-year agreement but need to validate with RPA colleagues whether holds need to be marketing year specific.  If so then we can create a new ticket to update our holds capability in the service.